src/cloud-api-adaptor/pkg/securecomms/wnssh/wnssh.go

```diff
@@ -0,0 +1,386 @@
+	}
+
+	ppAddr := make([]string, len(ipAddr))
+	for i, ip := range ipAddr {
+		ppAddr[i] = ip.String() + ":" + sshutil.SSHPORT
+	}
+
+	ctx, cancel := context.WithCancel(ctx)
+	ci := &SshClientInstance{
+		sid:          sid,
+		publicKey:    serverSshPublicKeyBytes,
+		ppAddr:       ppAddr,
+		sshClient:    c,
+		ctx:          ctx,
+		cancel:       cancel,
+		inboundPorts: make(map[string]string),
+	}
+
+	for _, tag := range c.inboundStrings {
+		name, inPort, err := ci.inbounds.Add(tag, &ci.wg)
+		if err != nil {
+			log.Printf("failed to add inbound: %v", err)
+		}
+		ci.inboundPorts[name] = inPort
+	}
+	for _, tag := range c.outboundStrings {
+		ci.outbounds.Add(tag)
+	}
+
+	return ci
+}
+
+func (ci *SshClientInstance) Start() error {
+	if !ci.k8sPhase {
+		// Attestation Phase
+		log.Println("Attstation Phase: Starting")
+		if err := ci.StartAttestation(); err != nil {
+			return fmt.Errorf("attstation Phase failed: %v", err)
+		}
+		log.Println("Attstation Phase: Done")
+		ci.k8sPhase = true
+	}
+
+	// Kubernetes Phase
+	ci.wg.Add(1)
+	go func() {
+		defer ci.wg.Done()
+		restarts := 0
+		for {
+			select {
+			case <-ci.ctx.Done():
+				log.Printf("Kubernetes Phase: Done")
+				return
+			default:
+				log.Printf("Kubernetes Phase: Starting (Number of restarts %d)", restarts)
+				if err := ci.StartKubernetes(); err != nil {
+					log.Printf("Kubernetes Phase: failed: %v", err)
+				}
+				time.Sleep(time.Second)
+				restarts += 1
+			}
+		}
+	}()
+	return nil
+}
+
+func (ci *SshClientInstance) StartKubernetes() error {
+	ctx, cancel := context.WithCancel(ci.ctx)
+	defer cancel()
+	peer := ci.StartSshClient(ctx, sshproxy.KUBERNETES, ci.publicKey, ci.sid)
+	if peer == nil {
+
+		return fmt.Errorf("Kubernetes Phase: failed StartSshClient")
+	}
+
+	peer.AddOutbounds(ci.outbounds)
+	err := peer.AddInbounds(ci.inbounds)
+	if err != nil {
+		peer.Close("Inbounds failed")
+		peer = nil
+		return fmt.Errorf("inbounds failed: %w", err)
+	}
+	peer.Wait()
+	return nil
+}
+
+func (ci *SshClientInstance) StartAttestation() error {
+	ctx, cancel := context.WithCancel(ci.ctx)
+	defer cancel()
+	peer := ci.StartSshClient(ctx, sshproxy.ATTESTATION, nil, ci.sid)
+	if peer == nil {
+		return fmt.Errorf("Attestation Phase: failed StartSshClient")
+	}
+	peer.AddOutbounds(ci.outbounds)
+	err := peer.AddInbounds(ci.inbounds)
+	if err != nil {
+		peer.Close("Inbounds failed")
+		peer = nil
+		return fmt.Errorf("inbounds failed: %w", err)
+	}
+	peer.Wait()
+	if !peer.IsUpgraded() {
+		return fmt.Errorf("Attestation PHASE closed without being upgraded")
+	}
+	return nil
+}
+
+func (ci *SshClientInstance) StartSshClient(ctx context.Context, phase string, publicKey []byte, sid string) *sshproxy.SshPeer {
+	config := &ssh.ClientConfig{
+		HostKeyCallback: func(hostname string, remote net.Addr, key ssh.PublicKey) error {
+			if len(publicKey) == 0 {
+				log.Printf("%s Phase: ssh Client skip validating server's HOST KEY (%s) during attestation", phase, key.Type())
```

Comment:
Please be mindful of random capitalization and sentence structure in user-visible messages.

```suggestion
				log.Printf("%s phase: ssh client skipped validating server's host %s key during attestation", phase, key.Type())
```

I am not sure if this is "skipped" or "skipping"?

When I see `HOST KEY` in upper cases, I expect this to be the name of some variable or something special like that.